### ISSUE TITLE
Support for http files (VSCode REST Client / IntelliJ HTTP Client)

### ISSUE THREAD

**Issue by lhns** [usebruno/bruno#1537]
For the developers in our Company we found using http files with [VSCode REST Client](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) or [IntelliJ HTTP Client](https://www.jetbrains.com/help/idea/http-client-in-product-code-editor.html) the most vendor agnostic way to work with http requests. The file format is very close to the actual HTTP Protocol and is very nice to manage in a VCS like git. Variables make it easy for non-technical people to work with these files but it would be even better if they could use these files in a tool like Bruno.
I propose support for http files in Bruno to make it easy for non-technical people to work with these files.

**Comment by nishudar** [comment]
I totally agree, http files is the current standard

**Comment by mamosk** [comment]
I agree too: in my company we already use `.http` files which are versioned in our git repositories; **best** option would be support for `.http` files; another option would be a converter from `.http` to `.bru` files (if you want to create a _new_ standard)

**Comment by aristidebm** [comment]
Still waiting for this feature, for neovim users, there is this plugin https://github.com/mistweaverco/kulala.nvim, that I find very useful, you may be interested.

**Comment by gorillamoe** [comment]
Kulala-fmt already supports OpenAPI, Postman and Bruno to .http file conversion.

So you could basically work with Bruno and use Kulala-fmt to save it as .http then.

We're also actively working on Kulala for Desktop which should be like Bruno, but for .http files.

Nevertheless it would be awesome if Bruno would support the .http format.

**Comment by LordEliasTM** [comment]
Is this feature coming anytime soon?

**Comment by vikaspoddar** [comment]
Has anyone started the implementation for http file integration?

**Comment by TeaTotalin** [comment]
I vote "yes" also.  This would be awesome.

**Comment by dreygur** [comment]
I would like to work on this.

**Comment by daneorie** [comment]
> Kulala-fmt already supports OpenAPI, Postman and Bruno to .http file conversion.
> 
> So you could basically work with Bruno and use Kulala-fmt to save it as .http then.
> 
> We're also actively working on Kulala for Desktop which should be like Bruno, but for .http files.
> 
> Nevertheless it would be awesome if Bruno would support the .http format.

@gorillamoe, how is this coming along? I'd be interested in trying to contribute if that would be helpful!

**Comment by gorillamoe** [comment]
> > Kulala-fmt already supports OpenAPI, Postman and Bruno to .http file conversion.
> > So you could basically work with Bruno and use Kulala-fmt to save it as .http then.
> > We're also actively working on Kulala for Desktop which should be like Bruno, but for .http files.
> > Nevertheless it would be awesome if Bruno would support the .http format.
> 
> [@gorillamoe](https://github.com/gorillamoe), how is this coming along? I'd be interested in trying to contribute if that would be helpful!

We're working on a core lib, that is going to power all the kulala tools in the future, so kulala.nvim doesn't have to have it's own implementation and kulala desktop it's own and kulala-fmt and/or kulala-cli doing their own stuff as well.

This is the current progress so far:

PR: https://github.com/mistweaverco/kulala-core/pull/2
Issue with more fine-grained overview: https://github.com/mistweaverco/kulala-core/issues/1

After this is finished, it should be pretty easy to get a real beta for the Desktop app out quickly.

**Comment by gorillamoe** [comment]
Quick heads up:

Kulala-Core has been released around a month ago and all Kulala Tools have been migrated to use core now.

Kulala for vscode is currently in beta and Kulala for Desktop will be worked on again in the next few weeks.

The plan is to have a public beta for both flavours (vscode plugin and desktop standalone) around July/August 2026.

Kulala-fmt (the formatter and converter) has been updated as well and should have  better .bru-/OpenApi-/Swagger-/Postman-Conversion support now.

**Comment by gorillamoe** [comment]
I just updated the formatter (and converter) again today, since I found out that Bruno is not only using .bru anymmore (which broke conversion to .http).

Anyway, the conversion from .bru or Bruno opencollention.yml and bruno-request.yaml files should work again.

Afaik the output is not only compatible with the Kulala tools, but also with Jetbrains.

Secrets are "lost" in the conversion. ["Everything else"](https://github.com/mistweaverco/kulala-fmt/tree/main/test-files/bruno/kulala) gets translated;
which results in this `Kulala.http` file (taken from the bruno collection name, which is "Kulala"):

```http
### get-image-png

# Converted from Bruno: get-image-png

GET https://echo.kulala.app/image/png HTTP/1.1

### graphql-query-person

# Converted from Bruno: graphql-query-person

GRAPHQL https://swapi-graphql.netlify.app/graphql HTTP/1.1
Accept: application/json
Content-Type: application/json
X-REQUEST-TYPE: GraphQL

query Person($id: ID) {
  person(personID: $id) {
    name
  }
}

{ "id": 1 }

### grpc-postman-echo

# Converted from Bruno: grpc-postman-echo

GRPC grpc.postman-echo.com:443 HelloService/SayHello
Content-Type: application/json

{
  "greeting": "Hello from Kulala!"
}

### multipart-form-data

# Converted from Bruno: multipart-form-data

POST {{API_BASE}}/post HTTP/1.1
Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7n5um4lmsi

------WebKitFormBoundary7n5um4lmsi
Content-Disposition: form-data; name="name"
Content-Type: text

Kulala
------WebKitFormBoundary7n5um4lmsi
Content-Disposition: form-data; name="textfile"
Content-Type: file

testfile.txt
------WebKitFormBoundary7n5um4lmsi--

### simple-get-json

# Converted from Bruno: simple-get-json

GET {{API_BASE}}/get HTTP/1.1
Accept: application/json

### simple-post-json

# Converted from Bruno: simple-post-json

POST {{API_BASE}}/post HTTP/1.1
Accept: application/json
Authorization: Bearter {{API_TOKEN_SECRET}}
Content-Type: application/json

{
  "kulala": "family"
}

### websocket-echo

# Converted from Bruno: websocket-echo

WEBSOCKET wss://ws.ifelse.io
Content-Type: application/json

{
  "name": "Kulala"
}
```

Additionally the environment(s) are translated to the `http-client.*.json` files:

[`http-client.env.json`](https://kulala.app/usage/environments#http-clientenvjson) (compatible with Jetbrains)

```json
{
  "$schema": "https://kulala.app/http-client.env.schema.json",
  "prod": {
    "API_BASE": "https://echo.kulala.app"
  }
}
```

[`http-client.private.env.json`](https://kulala.app/usage/environments#http-clientenvjson) (compatible with Jetbrains)

```json
{
  "$schema": "https://kulala.app/http-client.private.env.schema.json",
  "prod": {
    "API_TOKEN_SECRET": ""
  }
}
```

**Comment by gorillamoe** [comment]
> > > Kulala-fmt already supports OpenAPI, Postman and Bruno to .http file conversion.
> > > So you could basically work with Bruno and use Kulala-fmt to save it as .http then.
> > > We're also actively working on Kulala for Desktop which should be like Bruno, but for .http files.
> > > Nevertheless it would be awesome if Bruno would support the .http format.
> > 
> > [@gorillamoe](https://github.com/gorillamoe), how is this coming along? I'd be interested in trying to contribute if that would be helpful!
> 
> We're working on a core lib, that is going to power all the kulala tools in the future, so kulala.nvim doesn't have to have it's own implementation and kulala desktop it's own and kulala-fmt and/or kulala-cli doing their own stuff as well.
> 
> This is the current progress so far:
> 
> PR: https://github.com/mistweaverco/kulala-core/pull/2
> Issue with more fine-grained overview: https://github.com/mistweaverco/kulala-core/issues/1
> 
> After this is finished, it should be pretty easy to get a real beta for the Desktop app out quickly.


The desktop app has been released a few hours ago.


Now, we should have a flavour for every taste:

- Neovim
- vscode
- Zed (not by us, but based on the Kulala tool chain)
- CLI
- Standalone Desktop

**Comment by helloanoop** [comment]
Thanks for the work on this @gorillamoe ❤️ ❤️

**Comment by helloanoop** [comment]
Unfortunately, we don't see a viable path toward supporting .http files.

Over the last five years of building Bruno, the product has evolved significantly. We took those learnings and designed OpenCollection, an open specification for API collections and workflows. You can learn more at https://www.opencollection.com/, and explore the full schema visually at https://schema.opencollection.com/.

Our custom .bru format is now deprecated, and all new collections are created using the OpenCollection YAML format.

At this point, the schema has grown far beyond what can be reliably serialized to and deserialized from the .http format without losing information. To give you an idea of the complexity, I've attached a screenshot showing the depth of the schema and its nesting.

<img width="1200"  alt="Image" src="https://github.com/user-attachments/assets/df7a2062-f47e-4509-ab30-13518f28a34c" />

That said, @gorillamoe's work on kulala-fmt (https://github.com/mistweaverco/kulala-fmt) should make it possible to convert simpler OpenCollection-based collections into the .http format. We think that's a great fit for a community-maintained project, and we'd be happy to see the `.http` parity need evolve in that direction.
